azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java

```diff
@@ -120,7 +122,9 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
     public Set<GrantedAuthority> convertGroupsToGrantedAuthorities(final List<UserGroup> groups) {
         // Map the authority information to one or more GrantedAuthority's and add it to mappedAuthorities
         final Set<GrantedAuthority> mappedAuthorities = groups.stream()
-                .filter(group -> aadTargetGroups.contains(group.getDisplayName()))
+                .filter(group -> aadAuthenticationProperties.getUserGroup().getAllowedGroups()
```

Comment:
the same as line 96, extract complicated condition to method.

Reply:
Updated